Reduce total occupancy

eringold opened the pull request:
Pull request overview
---------------------

Adjusts building occupancy schedules to closer align ComStock occupancy with US Commercial Building occupancy as determined from American Time Use Survey data.

[ComStock Change Document](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/Change%20Documentation/Changes/vTBD%20Occupancy/ComStock%20Update%20Total%20Occupancy.docx?d=wd2b82d92b392462ebdac9df0344f510a&csf=1&web=1&e=Qgk1pY)


### Pull Request Author

This pull request makes changes to (select all the apply):
 - [x] Workflow Measures

Author pull request checklist:
<!--- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->
 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [x] Both `options_lookup.tsv` files updated
 - [x] 10k+ test run
 - [x] Change documentation written
 - [x] ComStock documentation updated
 - [x] Added 'See ComStock License' language to first two lines of each code file

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a code review on GitHub
 - [x] All related changes have been implemented: data and method additions, changes, tests
 - [x] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] Reviewed change documentation
 - [x] Ensured code files contain License reference
 - [x] Results differences are reasonable
 - [x] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

eringold committed:
measure files

eringold committed:
enable measure in options lookup

eringold committed:
change measure to upgrade

eringold committed:
Merge branch 'main' of https://github.com/NREL/ComStock into eringold/occupant_adjustment

eringold committed:
update occ adjustment

eringold committed:
Merge branch 'main' of https://github.com/NREL/ComStock into eringold/occupant_adjustment

eringold committed:
fix merge

asparke2 commented on resources/measures/adjust_occupancy_schedule/measure.rb:
```diff
@@ -0,0 +1,126 @@
+    args << peak_occ_frac
+
+    # apply/not apply measure
+    apply_measure = OpenStudio::Ruleset::OSArgument.makeBoolArgument('apply_measure', true)
```
Why do we need an `apply_measure` argument? In ComStock, we would remove the `occupancy_schedule_adjust` input column from the buildstock.csv to take it out of the workflow. In other measures, this argument is vestigial from the days when ComStock was a PAT project and OS-server needed each measure to have at least one argument to turn on/off.

eringold replied:
I'll remove.

asparke2 commented on resources/measures/adjust_occupancy_schedule/resources/os_lib_schedules.rb:
```diff
@@ -0,0 +1,933 @@
+# *******************************************************************************
```
Are the method used from this part of openstudio-standards yet? If so, use from there and remove resource file if possible.

eringold replied:
I was [tracking](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/openstudio-standards%20refactor%20changes/openstudio-standards%20refactor%20change%20list.docx?d=w94e52e177b154a719fc3cae4e51fd6b4&csf=1&web=1&e=toT71v) this as a revision to make once the whole standards refactor is complete and released, but I think the methods used have already been implemented and could possibly be updated now.

mdahlhausen replied:
I agree with @asparke on this.
There may be an issue updating from this resource file to the new schedules methods.
I suggest copying these openstudio-standards schedule methods directly into the top of the measure.rb. It'll make it much easier to change them out in the refactor, and we won't have any surprises in them behaving differently:
```
- schedule_get_min_max
- schedule_ruleset_simple_value_adjust
```

asparke2 reviewed (changes requested):
Minor code changes. Comment in the word doc RE calculating the % reduction fraction; did it account for the fact that ComStock only models ~66% of the floor area?

eringold committed:
remove apply_measure arg

eringold committed:
remove apply_measure arg from options_lookup

mdahlhausen commented on resources/measures/adjust_occupancy_schedule/measure.rb:
```diff
@@ -0,0 +1,113 @@
+      sch_name = h[:name]
+      schedule = model.getScheduleRulesetByName(sch_name).get
+      # append reduced percentage to schedule name
+      schedule.setName(sch_name + " #{peak_occ_frac * 100}%")
```
may need a round or limit method so you don't make a schedule name like "my old schedule 23.11239394938400204324%"

mdahlhausen replied:
also make the name descriptive, like "reduced by" or "x% of sch"

mdahlhausen reviewed (changes requested):
The change documentation looks great - thanks for including all the comparisons!
A few minor changes:
- [x] Copy methods from openstudio-standards instead of using the methods in os_lib_schedules.rb. That will make it easier to refactor in the future.
- [x] Update the workflow measures [README](https://github.com/NREL/ComStock/blob/main/resources/measures/README.md) with this measure.
- [x] Add ComStock license to files.
- [x] Update ComStock documentation in the occupancy section to explain this update.

eringold committed:
update measure and test:
-remove oslib_schedules
-copy new standards methods to be replaced
-descriptive schedule name with rounded value
-test runs, output the same
-add copyright
-update measure xml

eringold committed:
update workflow measure readme

eringold committed:
update documentation

eringold commented:
@asparke2 @mdahlhausen Thanks for the review. I think I've addressed all comments, let me know if there's anything else you'd like to see, particularly on the documentation.

mdahlhausen reviewed (approved):
Looks excellent!

asparke2 reviewed (approved):
Looks good, let's merge
